diff -p -up Squeak-4.0.3.2202-src/unix/config/configure.ac.orig Squeak-4.0.3.2202-src/unix/config/configure.ac
--- Squeak-4.0.3.2202-src/unix/config/configure.ac.orig	2010-10-09 01:15:51.000000000 -0300
+++ Squeak-4.0.3.2202-src/unix/config/configure.ac	2010-10-09 01:16:55.000000000 -0300
@@ -85,8 +85,8 @@ AC_SUBST(VM_VERSION)
 # libdir contains ${exec_prefix}, so we have to default and expand early
 test "x$prefix" = xNONE && prefix=$ac_default_prefix
 test "x$exec_prefix" = xNONE && exec_prefix='${prefix}'
-imgdir=`eval echo ${libdir}/squeak`
-plgdir=`eval echo ${imgdir}/${VM_VERSION}`
+imgdir=`eval echo ${datadir}/squeak`
+plgdir=`eval echo ${libdir}/squeak/${VM_VERSION}`
 
 AC_SUBST(imgdir)
 AC_SUBST(plgdir)
@@ -301,11 +301,7 @@ fi
 
 # Install locations
 
-if test "x${prefix}" = xNONE; then
-  SQ_LIBDIR=${ac_default_prefix}/lib/squeak/${SQ_VERSION}
-else
-  SQ_LIBDIR=${prefix}/lib/squeak/${SQ_VERSION}
-fi
+SQ_LIBDIR=${libdir}/squeak/${SQ_VERSION}
 
 AC_SUBST(SQ_LIBDIR)
 AC_DEFINE_UNQUOTED(SQ_LIBDIR,["${SQ_LIBDIR}"])
diff -p -up Squeak-4.0.3.2202-src/unix/config/configure.orig Squeak-4.0.3.2202-src/unix/config/configure
--- Squeak-4.0.3.2202-src/unix/config/configure.orig	2010-10-09 01:14:42.000000000 -0300
+++ Squeak-4.0.3.2202-src/unix/config/configure	2010-10-09 01:15:39.000000000 -0300
@@ -1604,8 +1604,9 @@ _ACEOF
 # libdir contains ${exec_prefix}, so we have to default and expand early
 test "x$prefix" = xNONE && prefix=$ac_default_prefix
 test "x$exec_prefix" = xNONE && exec_prefix='${prefix}'
-imgdir=`eval echo ${libdir}/squeak`
-plgdir=`eval echo ${imgdir}/${VM_VERSION}`
+imgdir=`eval echo ${datadir}/squeak`
+squeaklibdir=`eval echo ${libdir}/squeak`
+plgdir=`eval echo ${squeaklibdir}/${VM_VERSION}`
 
 
 
@@ -24431,11 +24432,7 @@ fi
 
 # Install locations
 
-if test "x${prefix}" = xNONE; then
-  SQ_LIBDIR=${ac_default_prefix}/lib/squeak/${SQ_VERSION}
-else
-  SQ_LIBDIR=${prefix}/lib/squeak/${SQ_VERSION}
-fi
+SQ_LIBDIR=${libdir}/squeak/${SQ_VERSION}
 
 
 cat >>confdefs.h <<_ACEOF
