diff -p -up Squeak-4.0.3.2202-src/unix/config/inisqueak.in.orig Squeak-4.0.3.2202-src/unix/config/inisqueak.in
--- Squeak-4.0.3.2202-src/unix/config/inisqueak.in.orig	2010-10-09 01:18:00.000000000 -0300
+++ Squeak-4.0.3.2202-src/unix/config/inisqueak.in	2010-10-09 01:18:22.000000000 -0300
@@ -182,7 +182,7 @@ if ${list}; then
       echo
     fi
   done
-  IMAGE=`echo ${images} | tr ' ' '\012' | tail +${reply} | head -1`
+  IMAGE=`echo ${images} | tr ' ' '\012' | tail -n +${reply} | head -1`
   CHANGES=${IMAGE}.changes.gz
   IMAGE=${IMAGE}.image.gz
 fi
